pyiron_base/project/generic.py

```diff
@@ -1514,6 +1513,33 @@ def unpack(self, origin_path, csv_file_name='export.csv', compress=True):
             self, archive_directory=origin_path, df=df, compressed=compress
         )
 
+    @classmethod
+    def register_tools(cls, name: str, creator):
+        """
+        Add a new creator to the project class.
+
+        Example)
+        >>> from pyiron_base import Project, Toolkit
+        >>> class MyCreator(Toolkit):
+        >>>     @property
+        >>>     def foo(self):
+        >>>         return 'foo'
+        >>>
+        >>> Project.register_tools('my_creator', MyCreator)
+        >>> pr = Project('scratch')
+        >>> print(pr.my_creator.foo)
+        'foo'
+
+        The intent is then that pyiron submodules (e.g. `pyiron_atomistics`) define a new creator and in their
+        `__init__.py` file only need to invoke `Project.register_creator('pyiron_submodule', SubmoduleCreator)`.
+        Then whenever `pyiron_submodule` gets imported, all its functionality is available on the project.
+
+        Args:
+            name (str): The name for the newly registered property.
+            creator (Toolkit): The creator to register.
+        """
+        setattr(cls, name, property(lambda self: creator(self)))
```

Comment:
Should probably do a `hasattr` check, so that we're not overwriting things without knowing.